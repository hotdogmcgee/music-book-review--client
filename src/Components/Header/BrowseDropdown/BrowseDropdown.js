import React from "react";
import { Link } from 'react-router-dom'
import './BrowseDropdown.css'

//want to make ul items clickable on full box, not just text area.
export default class BrowseDropdown extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displayMenu: false
    };

    this.showDropdownMenu = this.showDropdownMenu.bind(this);
    this.hideDropdownMenu = this.hideDropdownMenu.bind(this);
  }

  showDropdownMenu(event) {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener("click", this.hideDropdownMenu);
    });
  }

  hideDropdownMenu() {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener("click", this.hideDropdownMenu);
    });
  }

  render() {
    return (
      <div className="dropdown" style={{ background: "red", width: "200px" }}>
        <div className="button" onClick={this.showDropdownMenu}>
          
          Browse
        </div>

        {this.state.displayMenu ? (
          <ul>
            <li>
              <Link className="active" to="/category/instrument">

                Instrument

              </Link>
            </li>
            <li>
              <Link to="/category/author">Author</Link>
            </li>
            <li>
              <Link to="/category/publisher">Publisher</Link>
            </li>
            <li>
              <Link to="/category/average-rating">Average Rating</Link>
            </li>

          </ul>
        ) : null}
      </div>
    );
  }
}
