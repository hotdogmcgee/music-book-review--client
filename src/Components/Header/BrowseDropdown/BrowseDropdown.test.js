import React from 'react';
import BrowseDropdown from './BrowseDropdown'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('BrowseDropdown component', () => {

    it('renders a BrowseDropdown by default', () => {
        const wrapper = shallow(<BrowseDropdown />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})