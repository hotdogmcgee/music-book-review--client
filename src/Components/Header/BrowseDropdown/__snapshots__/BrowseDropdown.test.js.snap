// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseDropdown component renders a BrowseDropdown by default 1`] = `
<div
  className="dropdown"
>
  <div
    className="button"
    onClick={[Function]}
  >
    Browse
  </div>
</div>
`;
